builtin/game/misc.lua

```diff
@@ -6,6 +6,11 @@ local S = core.get_translator("__builtin")
 -- Misc. API functions
 --
 
+-- Checks whether the given string is a valid player name (see src/player.h)
+function core.is_valid_player_name(name)
+	return #name >= 1 and #name <= 20 and not name:find"[^a-zA-Z0-9_%-]"
```

Comment:
this should rather be C++ lest it gets out-of-sync
(also what's the relation to this PR?)

Reply:
> this should rather be C++ lest it gets out-of-sync

Will do.

> also what's the relation to this PR?

See https://github.com/minetest/minetest/pull/13987#discussion_r1703332582

(Note: I'd leave this as a separate commit when manually squashing this PR).

Reply:
I see, makes sense.